vitest.extend.ts

```diff
@@ -25,13 +24,91 @@ function sanitizeSVG(value: string) {
 }
 
 export const roundSnapshotOptions = {
-  cloneDeepWith: (value: any) => {
+  cloneDeepWith: (value: unknown) => {
     if (typeof value === 'number') {
       return Math.round(value);
     }
   },
 };
 
+const rawToMatchSnapshot = (chai.Assertion.prototype as any).toMatchSnapshot;
+
+chai.util.addMethod(
+  chai.Assertion.prototype,
+  'toMatchObjectSnapshot',
+  function (
+    this: Chai.AssertionStatic,
+    {
+      cloneDeepWith: customiser,
+      includeDefaultValues,
+      ...properties
+    }: ObjectOptions = {},
+    hint?: string,
+  ) {
+    let snap: Record<string, unknown>;
+    const received = chai.util.flag(this, 'object');
+
+    if (received instanceof FabricObject) {
+      const restore = received.includeDefaultValues;
+      if (typeof includeDefaultValues === 'boolean')
+        received.includeDefaultValues = includeDefaultValues;
+      snap = received.toObject();
+      received.includeDefaultValues = restore;
+    } else {
+      snap = received;
+    }
+    delete snap.version;
+
+    const value = cloneDeepWith(snap, (v, k, obj, stack) => {
+      const c = customiser?.(v, k, obj, stack);
+      if (c !== undefined) return c;
+      if (k === 'width') return Math.round(v as number);
+    });
```

Comment:
I Need to revisit why we need this.
Is true that fabricJS can output floats that could behave differently between node and browser for strange reasons, or just between platforms.
But if we need to round to have stable snapshot we need to use 'toFixed' not a generic round, and we need to use it on all numeric properties.

For now is fine as it is.

Reply:
Well this is just a move to pure vitest to not rely on 3rd party jest package.
I did not change logic at all and this is from jest times.
If you can improve or simplify this it can be a different PR.
I did this to drop the 3rd party package because we use vitest and not jest anymore and also to not block future tests in browser more.
A big bonus is that we install less packages now, lock file is smaller by thousand lines